packages/react-components/src/components/time-sync/viewportAdapter.tsx

```diff
@@ -0,0 +1,122 @@
+import parse from 'parse-duration';
+import type { DateRangePickerProps } from '@cloudscape-design/components';
+import type { Viewport } from '@iot-app-kit/core';
+
+const relativeOptionKey = (amount: number, unit: DateRangePickerProps.TimeUnit): string =>
+  `previous-${amount}-${unit}s`;
+
+const relativeOption = (amount: number, unit: DateRangePickerProps.TimeUnit): DateRangePickerProps.RelativeOption => ({
+  key: relativeOptionKey(amount, unit),
+  amount,
+  unit,
+  type: 'relative',
+});
+
+export const relativeOptions: DateRangePickerProps.RelativeOption[] = [
+  relativeOption(1, 'minute'),
+  relativeOption(5, 'minute'),
+  relativeOption(10, 'minute'),
+  relativeOption(30, 'minute'),
+  relativeOption(1, 'hour'),
+  relativeOption(1, 'day'),
+  relativeOption(7, 'day'),
+  relativeOption(30, 'day'),
+  relativeOption(90, 'day'),
+];
+
+export const dateRangeToViewport = (value: DateRangePickerProps.Value): Viewport => {
+  if (value.type === 'relative') return { duration: `${value.amount} ${value.unit}` };
+  return {
+    start: new Date(value.startDate),
+    end: new Date(value.endDate),
+  };
+};
+
+export const viewportToDateRange = (viewport?: Viewport): DateRangePickerProps.Value | null => {
+  if (!viewport) return null;
+
+  if ('duration' in viewport) {
+    const duration = viewport.duration;
+
+    let amount: number | undefined = undefined;
+    let unit: DateRangePickerProps.TimeUnit | null = null;
+
+    if (typeof duration === 'string') {
+      /**
+       * This regex is taken from the parse-duration library
+       * We need to identify if there are multiple durations included in the string.
+       * If there are, we can infer that the duration was not created via the dashboard and must convert it into
+       * some unit the dashboard can represent. For this scenario we are using seconds because it is the smallest
+       * configurable unit on the dashboard.
+       */
+      const matches = duration.match(/(-?(?:\d+\.?\d*|\d*\.?\d+)(?:e[-+]?\d+)?)\s*([\p{L}]*)/giu);
+      if (!matches) return null;
+
+      if (matches.length > 1) {
+        // this is a custom duration that was not set by the dashboard
+        amount = parse(duration, 's');
+        unit = 'second';
+      } else {
+        // this was set by the dashboard or is easily mapped to an option in the dashboard
+        const [d] = matches;
+
+        const seconds = /(\d+)\s*ns|μs|ms|s/;
+        const minutes = /(\d+)\s*m/;
+        const hours = /(\d+)\s*h/;
+        const days = /(\d+)\s*d/;
+        const weeks = /(\d+)\s*w/;
+        const months = /(\d+)\s*month/;
+        const years = /(\d+)\s*y/;
```

Comment:
Do you have a useful reference available on working with time in regex? Might be useful to include it in the code.